### PR TITLE
Onboarding: update site topic suggestions styling

### PR DESCRIPTION
**Pull request by keoshi**
### Changes proposed in this Pull Request

* Onboarding: update site topic suggestions styling introduced in https://github.com/Automattic/wp-calypso/pull/31405:
  * Add focus styles to site topic suggestions.
  * Line up site topic suggestions with placeholder.

#### dotcom:
* Go to `/start/onboarding/site-topic`.
* Make sure everything lines up and that focus styles look good.

#### Jetpack:
* Go to `/jetpack/connect/site-topic/SITE_URL` where `SITE_URL` is an existing Jetpack site. Alternatively, you can [create a new site](https://jurassic.ninja/create/?shortlived&jetpack-beta) and connect it to go through the onboarding flow.
* On the site topic step, make sure everything lines up and that focus styles look good.

### Before

image

image

image

### After

image

image

image

### CODE REVIEW

File: client/jetpack-connect/style.scss [hunk 1]
```diff
@@ -936,6 +936,13 @@ $colophon-height: 50px; // wpcomColophon element at the bottom
 		background-color: var( --color-white );
 		border-radius: 3px;
 	}
+
+	.site-verticals-suggestion-search__topic-list-item {
+
```
Comment (tyxla) [inline client/jetpack-connect/style.scss]: Nit: I usually refrain from newlines in the beginning/end of a nested block.

**Comment by shaunandrews** [comment]
Very nice. 🚢

File: client/jetpack-connect/style.scss (hunk 1 above)
Comment (keoshi) [reply client/jetpack-connect/style.scss]: Oh, I _always_ do it in SCSS so it's clear they don't have any direct styles applied to them. 🙃

Comment (keoshi) [reply client/jetpack-connect/style.scss]: I'm happy to change this and it shouldn't hold us back. Do you have any strong opinions on either, @shaunandrews ?

Comment (shaunandrews) [reply client/jetpack-connect/style.scss]: Same as @tyxla, I never do the empty line — its worth noting you can remove the nesting here altogether by just adding the focus selector to the parent selector. `.site-verticals-suggestion-search__topic-list-item:focus`

Comment (tyxla) [reply client/jetpack-connect/style.scss]: Anyway, this is really too minor to hold this PR - feel free to ship as-is.

Comment (keoshi) [reply client/jetpack-connect/style.scss]: That's what it compiles to anyway, @shaunandrews, and the way I see it code organization and readability are two of the main reasons why we even use SASS/SCSS. That's one of the things I always do as well: nest pseudo selectors to improve their readability for whoever sees this file next. See [this reference](https://sass-guidelin.es/#exceptions), but I guess it comes down to personal preference once again.

I'll clean up the code and merge!

**Pull request merged**